Sources/GRPCCore/Call/Client/ClientRPCExecutionConfigurationCollection.swift

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2023, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+@available(macOS 13.0, iOS 16.0, watchOS 9.0, tvOS 16.0, *)
+public struct ClientRPCExecutionConfigurationCollection: Sendable {
+  private var elements: [MethodDescriptor: ClientRPCExecutionConfiguration]
+  private let defaultConfiguration: ClientRPCExecutionConfiguration
+
+  public init(defaultConfiguration: ClientRPCExecutionConfiguration) {
```

Comment:
We should default the default configuration so that it has no timeout and no execution policy.